src/main.rs

```diff
@@ -155,31 +172,26 @@ fn main() -> anyhow::Result<()> {
                         "Parsing",
                         format_args!("{crate_name} v{version} (current)"),
                     )?;
+
                     let rustdoc_path = rustdoc_cmd.dump(manifest_path, None, true)?;
+                    let current_crate = load_rustdoc(&rustdoc_path)?;
+
+                    // Can overwrite rustdoc of current crate.
+                    // For example, this happens when target-dir is specified in `.cargo/config.toml`.
+                    // That's the reason why we're immediately loading the rustdocs to memory.
                     let baseline_path = loader.load_rustdoc(
                         &mut config,
                         &rustdoc_cmd,
                         crate_name,
                         Some(version),
                     )?;
-                    rustdoc_paths.push((crate_name.clone(), baseline_path, rustdoc_path));
-                }
-                rustdoc_paths
-            };
-            let mut success = true;
-            for (crate_name, baseline_path, current_path) in rustdoc_paths {
-                let baseline_crate = load_rustdoc(&baseline_path)?;
-                let current_crate = load_rustdoc(&current_path)?;
+                    let baseline_crate = load_rustdoc(&baseline_path)?;
 
-                if !run_check_release(&mut config, &crate_name, current_crate, baseline_crate)? {
-                    success = false;
+                    run_check_release(&mut config, crate_name, current_crate, baseline_crate)?;
                 }
             }
-            if success {
-                std::process::exit(0);
-            } else {
-                std::process::exit(1);
-            }
+
+            std::process::exit(i32::from(!success));
```

Comment:
The new code is more clever, but much less readable and much harder to understand. The cleverness is hurting more than helping here. This is not a good tradeoff.

Reply:
I agree. But it's not really my fault.

The refactor I had in mind is `std::process::exit(if success { 0 } else { 1 });` which, in my opinion, is a bit more readable than the original code -- it makes it more clear that what is important is exiting with some error code, not branching through the possible states of `success`.

But `clippy` has suggested changing `if success { 0 } else { 1 }` to `i32::from(!success)` and I'm also not convinced it is more readable. Assuming we don't want to ignore the clippy lints, the only options are the old code and the code I proposed in this PR.

What do you think: should I somehow ignore the clippy lint and do the inlined `if`, or should I revert to the old code?

Reply:
I'd probably leave the old code as-is. Duplicating the `std::process::exit()` call feels like "the lesser evil" here, from a risk and readability perspective.

Reply:
Done in 5e811c3cf1c4076c49994b806421561739ecd9cf.